app.go

```diff
@@ -474,9 +474,32 @@ func (app *app) loop() {
 	}
 }
 
-type cleanFunc func()
+func (app *app) runCmdSync(cmd *exec.Cmd, pause_after bool) {
+	app.nav.previewChan <- ""
+	app.nav.dirPreviewChan <- nil
 
-var noopCleanFunc = func() {}
+	if err := app.ui.suspend(); err != nil {
+		log.Printf("suspend: %s", err)
+	}
+	defer func() {
+		if err := app.ui.resume(); err != nil {
+			app.quit()
+			os.Exit(3)
+			return
+		}
+	}()
+	defer app.nav.renew()
+	defer app.ui.loadFile(app, true)
+
+	err := cmd.Run()
+	if pause_after {
+		anyKey()
+	}
+
+	if err != nil {
+		app.ui.echoerrf("running shell: %s", err)
+	}
```

Comment:
I think the order of `app.ui.echoerrf` and `anyKey` doesn't matter since the former just updates the error message contents in the UI. In fact the original code uses the reverse order. If that's the case you could simplify the code to:

```go
if err := cmd.Run(); err != nil {
	app.ui.echoerrf("running shell: %s", err)
}

if pause_after {
	anyKey()
}
```

Reply:
Yep, this seems to work, good idea!